APP/Back/EVChargersAPI/EVChargersAPI/UserManagement/Controllers/UserController.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Data.Entities;
+using EVChargersAPI.UserManagement.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EVChargersAPI.UserManagement.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class UserController : ControllerBase
+    {
+        private readonly IUserService _userService;
+
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<User>>> GetAll()
+        {
+            IEnumerable<User> users = await _userService.GetAll();
+            return Ok(users);
+        }
+        [HttpPost]
+        public async Task<ActionResult<User>> Add(User user)
+        {
+            User createdUser = await _userService.Create(user);
```

Comment:
I'm not familiar with .NET, but I would say this is a similar concept to Scala's language. 

I've investigated a little bit - can you check [this](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/chaining-tasks-by-using-continuation-tasks) article? Also, [this](https://stackoverflow.com/questions/38660747/c-sharp-transform-an-async-task-from-one-type-to-another) was my entry point to discussion.